sotodlib/site_pipeline/analyze_bright_ptsrc.py

```diff
@@ -63,71 +48,77 @@ def get_xieta_src_centered(ctime, az, el, data, sso_name, threshold=5):
 
     csl = so3g.proj.CelestialSightLine.az_el(ctime, az, el, weather="typical")
     q_bore = csl.Q
-    
-    ### Signal strength criterion - a chance to flag detectors (should be extra rare)
+
+    # Signal strength criterion - a chance to flag detectors (should be extra
+    # rare)
     peaks, _ = scipy.signal.find_peaks(data, height=threshold * np.std(data))
     if not list(peaks):
         return
 
-    ## The mean planet position is used
-    dt = datetime.fromtimestamp(np.mean(ctime))
-    dt_str = dt.strftime("%Y/%m/%d %H:%M:%S")
-    obj = getattr(ephem, sso_name)()
-    obj.compute(dt_str)
-    ra_obj = obj.ra
-    dec_obj = obj.dec
-    q_obj = quat.rotation_lonlat(ra_obj, dec_obj)
-
-    ### rotate coord system around max point
+    # Need the planet position at the peak
     src_idx = np.where(data == np.max(data[peaks]))[0][0]
-    q_bs_t = coords.quat.quat(*np.array(q_bore)[src_idx])
-    xi0, eta0, _ = quat.decompose_xieta(~q_bs_t * q_obj)
-    q_det = quat.rotation_xieta(xi0, eta0)
-    q_total = ~q_det * ~q_bore * q_obj
+    d1_unix = ctime[src_idx]
+    planet = planets.SlowSource.for_named_source(sso_name, d1_unix * 1.)
```

Comment:
I assume this part of the change is to use the `planets` module within astropy?

Reply:
No, this is actually the sotodlib.coords.planets module that is imported at the beginning. I think the main change in this part is to calculate the planet position when crossing the detector of interest (at the highest amplitude peak) other than just taking the average over the scan time. This corrects for the ~0.3 degree elevation offset we were seeing in the past from Jupiter simulations.